nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/ReductionOpValidation.java

```diff
@@ -1058,4 +1060,168 @@ public void testReductionsBackwards() {
             sd.execBackwards(Collections.emptyMap());
         }
     }
+
+    @Test
+    public void testDotProductAttention(){
```

Comment:
As noted in earlier comment - maybe let's improve all of these with loop testing c/f order input arrays, at least; testing weird strides/permuted would be even better.